### PR TITLE
Register external app folders opened via "Open app" on the /apps hub (D297)

### PR DESCRIPTION
**Pull request by vasu2001**
## What

The explorer's "Open app" button (pane strip + shut-pane fallback) now records the open before navigating:

- **Workspace folder** → `app_recents.json`, exactly as before.
- **External folder** → registered in a new `~/.fused-render/registered_apps.json`, so it appears on the **/apps hub** under the virtual `linked` tag, with its own last-opened recency, `metadata.json` category, `preview.png` and `<title>` — all via the shared `app_dict` shape.

Opening IS registering — no link/unlink lifecycle. The registry stores `{path, openedAt}` (deduped by path, capped at 200), so it doubles as the external apps' recents; a hub card click posts the same endpoint and refreshes `openedAt` for free.

## Why fresh code, not a resurrection

`linked_apps.py` (deleted in #525) served consumers that stay dead: the `app` template mode, `_app` sentinel, link/status routes, and the `FUSED_RENDER_LINKED_APPS` env export. Nothing outside the /apps listing consumes this store. Its lessons are kept — registry over symlink, skip-don't-delete reads, corrupt file reads empty, workspace/ancestor entries filtered on read AND write — plus one the old module lacked: `MountGuard().blocks()` before any per-entry syscall (external paths are more likely to be mounts than workspace paths).

Recorded as **D297** in DECISIONS.md, explicitly scoped as a deliberate partial revival of what D264/#525 removed, on the owner's ask.

## Tests

- New `tests/test_registered_apps.py` (12 tests): registration, re-open updates openedAt, workspace/ancestor refusal, skip-don't-delete, corrupt registry, category, merge with workspace apps, cap.
- Full suite: 6738 passed; the 2 failures are pre-existing env issues on macOS (`/proc/self/fd` Linux-only in test_calls, joblib template dep per D276).
- Frontend: tsc clean, 918 tests pass.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

---

> [!NOTE]
> **Low Risk**
> Adds a local JSON registry and extends an existing API with validation and mount guards; no auth or payment paths touched.
> 
> **Overview**
> **External app folders** opened from the explorer now show on the **/apps hub** under the virtual **`linked`** tag, with recency and the same card metadata as workspace apps — a scoped revival of the old linked-apps registry (D297), without template modes or link/unlink APIs.
> 
> **Open app** in the explorer (pane strip and shut-pane bar) shares one handler that **fire-and-forgets** `POST /api/apps/recents/open` with the **folder** path, then navigates to the entry file. Workspace opens still go to `app_recents.json`; paths outside the workspace are written to **`~/.fused-render/registered_apps.json`** via new **`registered_apps`**. Opening counts as registering (dedupe by path, cap 200, `openedAt` refresh on re-open).
> 
> **`GET /api/apps`** merges registered entries into the workspace walk; **`POST /api/apps/recents/open`** branches on workspace vs external. Registration is **server-validated** (absolute dir, has a page, not inside/over workspace, **`MountGuard`** before syscalls, **`realpath`** to block symlink aliases). Missing or bad targets are **skipped on read**, not deleted. **`tests/test_registered_apps.py`** covers the split and edge cases.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit c5a70c3148145954e613cf671192e51790c9801d. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: fused_render/registered_apps.py [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+    can't poison the listing either way."""
+    from fused_render.shell.seed import fused_dir
+
+    root = os.path.abspath(fused_dir())
+    folder = os.path.abspath(folder)
+    return (
+        root == folder
+        or root.startswith(folder + os.sep)
+        or folder.startswith(root + os.sep)
+    )
```
[the remark this answers is not in the document]
Comment (vasu2001) [reply fused_render/registered_apps.py:76]: Fixed in 7aa2399d: added a realpath check (`_resolves_into_workspace`) run in `record_open` and the listing loop, after MountGuard passes (a resolve is a syscall, so it cannot run in the guard-less `read_entries` filter). Pinned by `test_symlink_alias_of_workspace_is_refused_and_filtered`.

File: fused_render/registered_apps.py [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+    on the path."""
+    from fused_render.shell.seed import fused_dir
+
+    root = os.path.realpath(fused_dir())
+    folder = os.path.realpath(folder)
+    return (
+        root == folder
+        or root.startswith(folder + os.sep)
+        or folder.startswith(root + os.sep)
+    )
```
[the remark this answers is not in the document]
Comment (vasu2001) [reply fused_render/registered_apps.py:102]: Fixed in c5a70c31: the resolve is caught inside `_resolves_into_workspace` itself — OSError answers True, which both callers read as refuse/skip ("cannot tell reads as refuse"). Pinned by `test_unresolvable_path_is_refused_not_a_500`.
